### PR TITLE
chore: adds canonical url to tutorial view

### PR DESCRIPTION
**Pull request by kendallstrautman**
## "Ready for Review" checklist

- [x] The Vercel preview link has been added to this PR's description
- [ ] The Asana task has been added to this PR's description
- [ ] This PR's link has been added to the "📐 Pull Request" field on the Asana task
- [x] This Vercel preview link has been added to the "🔍 Preview" field on the Asana task
- [x] The description template has been filled in below

---

## Relevant links

- [Preview link](https://dev-portal-git-kstutorial-canonical-url-hashicorp.vercel.app/vault/tutorials/getting-started/getting-started-intro) 🔎
- [Asana task](https://app.asana.com/0/1199634971449915/1202016460311406) 🎟️

## What

This PR adds a canonical url tag and title tag to the tutorial view. 

## Why

Tutorials can belong to numerous collections, the canonical url will ensure that this duplication of pages does not affect SEO by setting the tutorial in the _default_ collection context is referenced as the canonical url. 

## Testing

- [Visit this page](https://dev-portal-git-kstutorial-canonical-url-hashicorp.vercel.app/vault/tutorials/getting-started/getting-started-intro) and check in `head` that the canonical url is set to this page
- [Visit another version](https://dev-portal-git-kstutorial-canonical-url-hashicorp.vercel.app/vault/tutorials/cloud/getting-started-intro) of this tutorial that isn't the default collection, check that the canonical url is still there and the same.

### CODE REVIEW

File: src/views/tutorial-view/utils/index.ts [hunk 1]
```diff
@@ -19,3 +19,14 @@ export function formatTutorialToMenuItem(
     isActive: path === currentPath,
   }
 }
+
+export function generateCanonicalUrl(
+  defaultCollectionSlug: string,
+  tutorialSlug: string
+): URL {
+  const path = getTutorialSlug(tutorialSlug, defaultCollectionSlug)
+  const base = 'https://developer.hashi-mktg.com'
```
Comment (kendallstrautman) [inline src/views/tutorial-view/utils/index.ts:28]: Perhaps we lift this up 🤔 I'm assuming this will change during the production release.

Comment (ashleemboyer) [reply src/views/tutorial-view/utils/index.ts:28]: Yeah, this seems like something that would be a good thing to put in the environment `/config` JSON files!

**Review by ashleemboyer — APPROVED** [review]
Checked the `<head>` of the two test pages you linked, works great! ✨

**Comment by kendallstrautman** [comment]
Ah that makes sense, I was all confused if `__config` was available at runtime. Maybe I'll adjust the name to `prod_base_url` since it may be misleading to have this on dev / preview configs as just `base_url`

**Pull request merged**